js/widgets/forms/textinput.js

```diff
@@ -81,57 +85,40 @@ $.widget( "mobile.textinput", {
 		});
 	},
 
-	_enhance: function() {
-		var elementClasses = [];
-
-		if ( this.isTextarea ) {
-			elementClasses.push( "ui-input-text" );
+	_addClasses: function() {
+		if ( this._searchIcon ) {
+			this._addClass( this._searchIcon, "ui-textinput-search-icon" );
 		}
+		this._addClass( this._outer,
+			"ui-textinput ui-textinput-" + ( this.isSearch ? "search" : "text" ),
+			"ui-body-" + ( this.options.theme ? this.options.theme : "inherit" ) );
+	},
 
-		if ( this.isTextarea || this.isRange ) {
-			elementClasses.push( "ui-shadow-inset" );
-		}
+	_enhance: function() {
+		var outer;
 
-		//"search" and "text" input widgets
 		if ( this.inputNeedsWrap ) {
-			this.element.wrap( this._wrap() );
+			outer = $( "<div>" );
+			if ( this.isSearch ) {
+				this._searchIcon = $( "<span>" ).appendTo( outer );
+				$( "<span> </span>" ).appendTo( outer );
```

Comment:
This should be `this._searchIcon.after( "<span> </span>" );` after fixing to `prepend` above.